### PR TITLE
Scripts: Upgrade Puppeteer to v11

### PR DESCRIPTION
**Pull request by gziolo**
## Description

This change is necessary to bring compatibility with npm 7 and 8. It's based on the work in #33892 (https://github.com/WordPress/gutenberg/pull/33892/commits/a78030a4c8f05b43139797360679b6893d3dbbb3). npm 7 and above by default installs all peer dependencies and that causes issues because `@wordpres/scripts` uses `puppeteer-core` instead of `puppeteer`. The solution is to make `puppeteer` an optional dependency for WordPress packages.

As part of the change, this PR also upgrades `puppeteer-core` to the latest version.

## How has this been tested?

`npm run test-e2e` still works locally and on CI.

## Types of changes
Breaking change for some packages and an enhancement for `@wordpress/scripts`.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by gziolo** [comment]
CI is now green so this should be good to go.

File: packages/e2e-test-utils/package.json [hunk 1]
```diff
@@ -39,7 +39,12 @@
 	},
 	"peerDependencies": {
 		"jest": ">=26",
-		"puppeteer": ">=1.19.0"
+		"puppeteer": ">=11"
+	},
+	"peerDependenciesMeta": {
+		"puppeteer": {
+			"optional": true
```
Comment (youknowriad) [inline packages/e2e-test-utils/package.json]: Is this the default behavior in npm7 or are we trying to keep the current behavior when we switch to npm7

**Comment by youknowriad** [comment]
> npm 7 and above by default installs all peer dependencies and that causes issues because @wordpres/scripts uses puppeteer-core instead of puppeteer. 

Reading this makes me wonder why we have a "puppeteer" dependency then? Should it be "puppeteer-core"? What's the difference?

File: packages/e2e-test-utils/package.json (hunk 1 above)
Comment (gziolo) [reply packages/e2e-test-utils/package.json]: Yes, it's to make sure it doens't install `puppeteer` in Gutenberg and in projects that use `@wordpress/e2e-test-utils` together with `@wordpress/scripts` that depends on `puppeteer-core`.

**Comment by gziolo** [comment]
> > npm 7 and above by default installs all peer dependencies and that causes issues because @wordpres/scripts uses puppeteer-core instead of puppeteer.
> 
> Reading this makes me wonder why we have a "puppeteer" dependency then? Should it be "puppeteer-core"? What's the difference?

Good point. For e2e test utils and e2e tests we can change it to `puppeteer-core`. I didn't think about the fact they will be used together with `@wordpress/scripts` anyway.

`@wordpress/jest-puppeteer-axe` was used as a dependency outside of WordPress so it's more likely to be used with `puppeteer` as a dependency so we might want to leave it as is.

**Review by youknowriad — APPROVED** [review]

**Pull request merged**